Explorer/Assets/DCL/Backpack/BackpackEquipStatusController.cs

```diff
@@ -142,9 +157,51 @@ private async UniTaskVoid UpdateProfileAsync(CancellationToken ct)
             bool publishProfileChange = !appArgs.HasFlag(AppArgsFlags.SELF_PREVIEW_BUILDER_COLLECTIONS)
                                         && !appArgs.HasFlag(AppArgsFlags.SELF_PREVIEW_WEARABLES);
 
-            var profile = await selfProfile.UpdateProfileAsync(publish: publishProfileChange, ct);
-            MultithreadingUtility.AssertMainThread(nameof(UpdateProfileAsync), true);
-            UpdateAvatarInWorld(profile!);
+            Profile? oldProfile = await selfProfile.ProfileAsync(ct);
```

Comment:
WDYT is we move the `oldProfile` logic to `SelfProfile`. 

I would name it something like `lastProfileSaved`. To me, makes sense that the logic is handled there:

- You can store which is the last profile that was successful so you don't have to rebuild it when a change is requested
- `ProfileBuilder` logic can exist only there and not here
- You can move the `if` conditions to that class, leaving this one a little bit more cleaner

Reply:
Will move everything, including the update of the ECS avatar in `SelfProfile`, but will do it in a different pr as it will involve a bigger refactor